### PR TITLE
flag to stop parsing options, fixes #433

### LINKED ISSUE #433: Add flag to cjxl/djxl to stop parsing options.
**Is your feature request related to a problem? Please describe.**
Filenames with leading `-` characters are hard to feed in and out of the CLI encoder/decoder as it tries to parse them as options.

**Describe the solution you'd like**
Add a special option (`--` seems to be pretty much ubiquitous for this) to tell the program to stop parsing further arguments as options.

**Describe alternatives you've considered**
There are workarounds, such as by specifying a full path (e.g `./-filename.jpg`), but this in inconvenient for use within scripts.

### PR DESCRIPTION
**Pull request by jonsneyers**
See #433. If for whatever reason you need things to work on filenames that start with a dash, you can now do `cjxl -q 80 -- -myweirdfile.png -myweirdoutputfile.jxl`.

### CODE REVIEW

File: tools/cmdline.h [hunk 1]
```diff
@@ -136,8 +136,8 @@ class CommandLineParser {
     // Only match non-flag values. This means that you can't pass '-foo' as a
```
Comment (deymo) [inline tools/cmdline.h:136]: This comment needs to be updated.

File: tools/cmdline.h [hunk 2]
```diff
@@ -41,7 +41,7 @@ class CommandLineParser {
     virtual bool matched() const = 0;
 
     // Returns whether this option matches the passed command line argument.
```
Comment (deymo) [inline tools/cmdline.h:43]: Please explain what's the meaning of `parse_options` in the comment.

**Review by deymo — APPROVED** [review]

File: tools/cmdline.h (hunk 1 above)
Comment (jonsneyers) [reply tools/cmdline.h:136]: Done.

File: tools/cmdline.h (hunk 2 above)
Comment (jonsneyers) [reply tools/cmdline.h:43]: Done.

**Review by deymo — APPROVED** [review]

**Pull request merged**